### DISCUSSION TITLE
Where can I find alternative implementation like SVM of BaseEstimator besides RotationForest and ContinuousIntervalTree?

### DISCUSSION THREAD

**Discussion by junshi356** [alan-turing-institute/sktime#3131] [Q&A]
Hi, I'm trying to use [ShapletTransformClassifier](https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.classification.shapelet_based.ShapeletTransformClassifier.html) to classify my time series data and I wish to use other implementations of estimator other than default Rotation Forest estimator. 

I tried to train a classifier under different combination of parameters like `n_shapelet_samples`, `max_shapelets` and `max_shapelet_length`, the classifier can't give a good enough predict accuracy on my test dataset. 

So I went to read the paper "**Binary Shapelet Transform for Multiclass Time Series Classification**" noted in document's reference part. It mentioned that after shapelet transformation, we can use the transformed data in conjunction with any classifier. It also listed several classifiers including WEKA, Naive Bayes, Support Vector Machines and Rotation Forest.

Rotation Forest is used as default estimator in sktime [code](https://github.com/alan-turing-institute/sktime/blob/32fdc4e5c18538af01e12300a975327e40775e0b/sktime/classification/shapelet_based/_stc.py#L42), I can only find "ContinuousIntervalTree" as another estimator inheriting `BaseEstimator` class under [sktime.classification.sklearn](https://github.com/alan-turing-institute/sktime/blob/32fdc4e5c18538af01e12300a975327e40775e0b/sktime/classification/sklearn/__init__.py#L1).

Can I have some advice on how to find other implementations of estimators that I can pass to `ShapletTransformClassifier`, or some workarounds to replace default Rotation Forest?

Thanks!

Section 2.1 of the paper discussing different classifiers after shapelet transformation.
![image](https://user-images.githubusercontent.com/40595015/181800137-110e2e7d-d506-4768-a403-2a4b815eee4d.png)

**Comment by fkiraly** [comment]
You can use any `sklearn` compatible estimator with the classifier, e.g.,

```python
from sktime.classification.shapelet_based import ShapeletTransformClassifier
from sklearn.svm import SVC
from sktime.utils._testing.panel import make_classification_problem

clf = ShapeletTransformClassifier(estimator=SVC())

X, y = make_classification_problem()

clf.fit(X, y)
```

you can find a lot in `sklearn`, and many more across other packages. Everything works as long as `sklearn` compatible.

**Comment by fkiraly** [comment]
The more general concept is a pipeline, this should work

```python
from sktime._contrib.shapelets import ShapeletTransform
from sktime.utils._testing.panel import make_classification_problem
from sklearn.svm import SVC

clf = ShapeletTransform() * SVC()

X, y = make_classification_problem()

clf.fit(X, y)
```

but it doesn't, yet, because the transformer has not been exposed by the authors to users, also see this issue: https://github.com/alan-turing-institute/sktime/issues/2942

Once that is addressed, it should be possible to use the shapelet transform more generally in pipelines.

**Reply by fkiraly** [reply]
This should now work with https://github.com/alan-turing-institute/sktime/pull/3136 (not merged/released yet)
